### DISCUSSION TITLE
mini.completion: Is there a way to not filter out LSP snippets?

### DISCUSSION THREAD

**Discussion by heygarrett** [nvim-mini/mini.nvim#517] [Q&A] [answered]
Currently I'm using the built-in omnicomplete, with `<c-x><c-o>`, and [nvim-snippy](https://github.com/dcampos/nvim-snippy) to expand LSP snippets. I understand that mini.completion doesn't support snippet expansion, but is there a way to include LSP snippets in the completion results so I can expand them with nvim-snippy?

**Comment by echasnovski** [comment, marked as answer]
Yes, there is a way, but it is not pretty.

Snippets are filtered out when processing LSP response before creating an actual list of completion entries. This part can be configured by `config.lsp_completion.process_items()`. But the thing is, there is no exported way which doesn't filter out snippets. But you can try something like this, which is a cherry-picked version of 'mini.completion' code:

```lua
local get_completion_word =
  function(item) return (item.textEdit or {}).newText or item.insertText or item.label or '' end

local process_items_with_snippets = function(items, base)
  local res = vim.tbl_filter(function(item) return vim.startswith(get_completion_word(item), base) end, items)
  table.sort(res, function(a, b) return (a.sortText or a.label) < (b.sortText or b.label) end)
  return res
end

require('mini.completion').setup({ lsp_completion = { process_items = process_items_with_snippets } })
```

**Reply by heygarrett** [reply]
Thanks! Do you have any plans to provide this as a configurable option? Maybe even if undocumented? 😅

**Reply by echasnovski** [reply]
Probably not until 'mini.snippets' is actually a thing. Which might be sooner than I hoped (but still not days/weeks ahead).
